### PR TITLE
Integrate llvm/llvm-project@27ac46e6bea2

### PR DESCRIPTION
**Pull request by antiagainst**
Updated to llvm/llvm-project@27ac46e6bea2
* Used LLVM `MathExtras.h` to replace MLIR one
* Updated `applySignatureConversion` usage

Updated to openxla/stablehlo@dd48ec5
* `chlo.minimum_broadcast_shapes` op was removed
  https://github.com/openxla/stablehlo/pull/2287
* `chlo.dynamic_reshape` op was removed
  https://github.com/openxla/stablehlo/pull/2286
* Added batching dims to scatter dims
  https://github.com/openxla/stablehlo/pull/2259

Updated to llvm/torch-mlir@77d7f64

### CODE REVIEW

**Review by kuhar — APPROVED** [review]

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
We need to look at the regressions in number of dispatches. I can help (but not today).

**Comment by qedawkins** [comment]
This seems like a fairly likely candidate for the source of dispatch count changes: https://github.com/iree-org/llvm-project/commit/7ef83f5561b34ca07fdef23ca2b3c01c583dbbf5

Especially because the changes are being observed in data tiling enabled benchmarks. cc @Max191

**Comment by nirvedhmeshram** [comment]
> We need to look at the regressions in number of dispatches. I can help (but not today).

@MaheshRavishankar are you blocking the integrate for this or would you look at it in a follow up since Quinn has explained the possible reason for the difference?

**Comment by MaheshRavishankar** [comment]
> > We need to look at the regressions in number of dispatches. I can help (but not today).
> 
> @MaheshRavishankar are you blocking the integrate for this or would you look at it in a follow up since Quinn has explained the possible reason for the difference?

Could you try reverting that locally to see if that is the issue. Then we can decide what to do next

**Comment by nirvedhmeshram** [comment]
> > > We need to look at the regressions in number of dispatches. I can help (but not today).
> > 
> > 
> > @MaheshRavishankar are you blocking the integrate for this or would you look at it in a follow up since Quinn has explained the possible reason for the difference?
> 
> Could you try reverting that locally to see if that is the issue. Then we can decide what to do next

@MaheshRavishankar PTAL at the benchmark comment now, the bot has edited it and it seems the dispatch number regression is gone with the revert.

**Review by MaheshRavishankar — APPROVED** [review]
Thanks!

File: compiler/setup.py [hunk 1]
```diff
@@ -464,6 +464,7 @@ def find_git_submodule_revision(submodule_path):
     install_requires=[
         "numpy",
         "PyYAML",
+        "sympy",
```
Comment (ScottTodd) [inline compiler/setup.py:467]: Looks like this new dep is for the `fx_importer` in torch-mlir. That should have at least been documented in the PR description.

This should probably go in `extras_require` for a `iree-compiler[torch]` package, not the general requirements here. The dep seems relatively small, but I want to keep the core project's dependencies minimal.

**Pull request merged**